common/src/main/java/org/opencds/cqf/ruler/common/utility/Canonicals.java

```diff
@@ -0,0 +1,151 @@
+package org.opencds.cqf.ruler.common.utility;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.hl7.fhir.instance.model.api.IIdType;
+import org.hl7.fhir.instance.model.api.IPrimitiveType;
+
+import ca.uhn.fhir.context.FhirVersionEnum;
+
+public class Canonicals {
+
+	private Canonicals() {
+	}
+
+	interface CanonicalParts<IdType extends IIdType> {
+		String getVersion();
+
+		String getUrl();
+
+		IdType getId();
+	}
+
+	public static <CanonicalType extends IPrimitiveType<String>> String getId(CanonicalType theCanonicalType) {
+		checkNotNull(theCanonicalType);
+		checkArgument(theCanonicalType.getValue() != null);
```

Comment:
These are examples of validating inputs to public APIs. Validate inputs as the very first step in a function.